src/Hydrators/DogHydrator.php

```diff
@@ -13,6 +13,17 @@ public static function getDogs(\PDO $db){
     $query->setFetchMode(\PDO::FETCH_CLASS, \Fetch\Classes\Dog::class);
     return $query->fetchAll();
     }
-}
 
+    /** Function to retrieve a single dog with all data from database
+     * @param \PDO $db takes a parameter that is the database of dogs
+     * @param string $dogID takes string containing dog ID from get request
+     * @return array containing a single dog object from database
+     */
+    public static function getSingleDog(\PDO $db, string $dogID){
```

Comment:
Could typehint the output perhaps?  Also while you're at it, could you also typehint the output of the method above this one?  It seems like Xandice and I forgot.